### ISSUE TITLE
Using with Jeditable

### ISSUE THREAD

**Issue by conorhub** [devbridge/jQuery-Autocomplete#709]
If unfamiliar with *Jeditable*, it dynamically changes text elements into text forms as a means of allowing a user to edit page text in place. It's common place to combine this with an *autocomplete* library (see the many questions on *stackoverflow*) and it provides plugin architecture to do so.

However, I'm getting a `JSON` error that I'm possibly misinterpreting when attempting so:

```
$.editable.addInputType('autocomplete', {
        element : $.editable.types.text.element,
        plugin : function(settings, original) {
            $('input', this).autocomplete(settings.autocomplete);
        }
    });

    $('.edit').editable('/services/organisesave.php', {
        type      : 'autocomplete',
        indicator : 'Saving...',
        event     : 'dblclick',
        submit    : 'Save',
        cssclass  : 'jeditactive',
        tooltip   : 'Double click to edit...',
        autocomplete : {
            source : ["Aberdeen", "Ada", "Adamsville", "Addyston", "Adelphi", "Adena", "Adrian", "Akron"],
            minChars : 2
        }
    });
```

Each keystroke reports a console error of:

`SyntaxError: JSON.parse: unexpected character at line 2 column 1 of the JSON data`

I added `dataType : 'text'`, with no change, and later `dataType : 'json'`, which removed the error but gave no result. This would make sense as I'm not trying to submit a `JSON` array. I'm wondering if the error is unrelated to the array I'm using as a source?

The format I'm following is similar to many used for combining several `autocomplete` plugins. See : https://stackoverflow.com/questions/1458404/working-example-of-jeditable-and-autocomplete-working-together#1458420 and I believe I have the correct settings in place.

<hr>

Update:
--------

Changed options to:
```

autocomplete : {
            "query" : "Unit",
            "suggestions" : ["Aberdeen", "Ada", "Adamsville", "Addyston", "Adelphi", "Adena", "Adrian", "Akron"],
            "minChars" : 2,
            "dataType" : 'text'
        }
```

After seeing I was too rash in thinking the `source` parameter was used, like similar plugins. I also included `query`, even though it states it's no longer needed, just to be thorough. Same result and same confusion over why I'm receiving a `JSON.Parse` error despite attempting to not send `JSON`, per the instructions here.

**Comment by tkirda** [comment]
There is no property `suggestions`.  You may try to change `suggestions` to `lookup`.

Looks like this is issue with setup and not autocomplete plugin. Also, are you sure you are using this autocomplete or `jQuery UI` autocomplete?

**Comment by conorhub** [comment]
@tkirda I took the source from this very github so am 100% I'm using this. 

Is your question relating to my usage of parameters relevant to *jQuery UI autocomplete*? I thought I'd addressed that when explaining I'd researched examples of *Jeditable* working with other *autocomplete* plugins. I thought I'd switched parameters to ones that were only relevant to this `autocomplete` when changing `source` to `suggestions` but, judging by your comment—perhaps not—maybe I'd left in remnants of properties others were using by accident though I can't see where.

I did get the `suggestions` property from your only example of using a non-data string array on this page: https://www.devbridge.com/sourcery/components/jquery-autocomplete/ where you use the property `suggestions`. I don't understand why you say it's not a property when it's the only string array example you yourself give?

**Comment by tkirda** [comment]
See here: https://github.com/devbridge/jQuery-Autocomplete#local-only-settings

It should be:
```javascript
lookup: ["Aberdeen", "Ada", "Adamsville"]
```
or
```javascript
lookup: [
    { value: "Aberdeen" },
    { value: "Ada" }
    // ...
]
```

**Comment by conorhub** [comment]
Thanks @tkirda that solved the issue and avoided resorting to JSON, which felt convoluted for an already existing, simpler array string. Aside from that, while it may be that I misinterpreted the relevance of the property `suggestion` (though your example on the your page I linked above explicity used it), I also don't understand why you'd say:

> There is no property suggestions

when it is the property listed one under the reference to `lookup` you just linked to? So I'm more confused than ever at the explanations but happy to have a working result. Thanks.
